coordinator/src/services/translation/incoming/release/release-response.ts

```diff
@@ -0,0 +1,106 @@
+import {ParentPrescription} from "../../../../models/hl7-v3/hl7-v3-prescriptions"
+import * as fhir from "../../../../models/fhir/fhir-resources"
+import {isDeepStrictEqual} from "util"
+import {convertResourceToBundleEntry, translateAndAddAgentPerson, translateAndAddPatient} from "../common"
+import {toArray} from "../../common"
+import {createMedicationRequest} from "./release-medication-request"
+import {createMessageHeader, EVENT_CODING} from "../message-header"
+import {createAndAddCommunicationRequest, parseAdditionalInstructions} from "./additional-instructions"
+import * as uuid from "uuid"
+import {PrescriptionReleaseResponse} from "../../../../models/hl7-v3/hl7-v3-release"
+import {convertHL7V3DateTimeToIsoDateTimeString} from "../../common/dateTime"
+
+const SUPPORTED_MESSAGE_TYPE = "PORX_MT122003UK32"
+
+export function createOuterBundle(releaseResponse: PrescriptionReleaseResponse): fhir.Bundle {
+  const releaseRequestId = releaseResponse.inFulfillmentOf.priorDownloadRequestRef.id._attributes.root
+  const parentPrescriptions = toArray(releaseResponse.component)
+    .filter(component => component.templateId._attributes.extension === SUPPORTED_MESSAGE_TYPE)
+    .map(component => component.ParentPrescription)
+  return {
+    resourceType: "Bundle",
+    id: uuid.v4(),
+    meta: {
+      lastUpdated: convertHL7V3DateTimeToIsoDateTimeString(releaseResponse.effectiveTime)
+    },
+    identifier: {
+      system: "https://tools.ietf.org/html/rfc4122",
+      value: releaseResponse.id._attributes.root.toLowerCase()
+    },
+    type: "searchset",
+    total: parentPrescriptions.length,
+    entry: parentPrescriptions
+      .map(parentPrescription => createInnerBundle(parentPrescription, releaseRequestId))
+      .map(convertResourceToBundleEntry)
```

Comment:
Rather than going through the array again to add the fullURL, could we not just add it in the createInnerBundle() function?

Reply:
Although thinking about it, it would make that function a bit messier

Reply:
Yeah I don't see a benefit. I prefer having these functions return FHIR resources instead of bundle entries anyway. Resources are useful in their own right, but bundle entries are just part of a bundle.